spec/solver_spec.rb

```diff
@@ -0,0 +1,36 @@
+require_relative '../solver'
+
+describe Solver do
+  before(:each) do
+    @solver = Solver.new
+  end
```

Comment:
- [ ] If a negative integer is given it should raise an exception. Kindly add a `#factorial` method for the negative value. :+1: